### ISSUE TITLE
Cannot cancel a donation of over 1 mil

### ISSUE THREAD

**Issue by q30a** [flathub/website#305]
If someone feels really generous and wants to donate > 1 mil and changes their mind halfway through, a bug occurs:
1. Go to https://beta.flathub.org/donate
2. Enter amount, in my case 1.5 mil
3. Press 'Make Donation'
4. Whoops! Error: Failed to load Stripe.js - happens for every attempt so I guess it's expected
5. Press 'Go to User Wallet'
6. Press 'Cancel Transaction' next to the generous donation

Current behavior:
400 response: `{"status":"error","error":"stripe-payment-intent-build-failed"}`

Expected behavior
We need some input validation on the frontend I think. Do not allow proceeding with bonkers values. When one is even more generous, then other bugs occur:
1. $22000000000000.00 yields 500 response `Internal Server Error` without any message
2. $999999999999999999999999 converts the input to scientific notation 1e+24 and the only way to delete it is by selecting predefined amount
3. If there is a minimal value then it should be validated as well or it is unclear to the user what happens:
https://user-images.githubusercontent.com/105132988/170796432-3b13654f-eed4-458b-871e-aac401e396c6.mov

**Comment by kymckay** [comment]
Thanks for testing this out and providing excellent feedback :smiley: 

- The Stripe.js error looks like we need to update frontend CSP to allow that script to load. I'll open an issue for that now.
- The payment intent failed error is interesting, we'll need to investigate what's going on with the backend there. Unseen in development so could be something about the staging environment causing this (incorrect Stripe set up perhaps).
- The current $5 minimum there is somewhat arbitrary until we establish desired constraints on donations/payment which is why not much thought went into validation. Let's track that in a new issue too.

**Comment by razzeee** [comment]
The errors hit sentry seem to suggest, that the database insert on our side fails. As the test value is bigger then the integer size https://www.postgresql.org/docs/current/datatype-numeric.html

**Comment by nikodunk** [comment]
I'm now no longer getting the "Failed to load Stipe.js" error and can successfully donate. Thanks for fixing.

**Comment by kymckay** [comment]
This is partially addressed in that the backend will now deny creation of transactions above 999,999.99 (the maximum Stripe can handle)

For now the frontend will just give a generic error message when that is encountered. In future once #353 is merged it will be trivial to add a limit to the input component which will prevent submission and give feedback if exceeded (making that error unreachable under normal circumstances).

As for these already existing transactions, they might just exist in an invalid state until the staging environment is reset. Though we may also want to update the transaction cancellation endpoint to consume those intent failed errors (they can occur for a few other reasons too) and just perform the cancellation.
